common/middleware.py

```diff
@@ -10,29 +10,39 @@ def __init__(self, get_response=None):
     def __call__(self, request):
         response = self.get_response(request)
         try:
-            res = json.loads(response.content.decode())
+            content = json.loads(response.content.decode())
         except json.JSONDecodeError:
             return response
 
-        if request.method == 'GET':
-            # Use case for retrieval of non-exitent record
-            if 'data' in res and list(res['data'].values())[0] is None:
-                res = None
-            # Use case for retrieval of all records
-            elif 'data' in res and 'edges' in list(res['data'].values())[0]:
-                res = list(list(res['data'].values())[0].values())[0]
-                res = [list(item.values())[0] for item in res]
-            # Use case for retrieval of a single existing record
-            elif 'data' in res and isinstance(list(res['data'].values())[0], dict):
-                res = list(res['data'].values())[0]
-        elif request.method == 'POST':
-            # Use case for creating, deleting and updating a record
-            if 'data' in res and isinstance(list(list(res['data'].values())[0].values())[0], dict):
-                res = list(list(res['data'].values())[0].values())[0]
-            # Use case for deleting and updating a record
-            elif 'data' in res and list(list(res['data'].values())[0].values())[0] is None:
-                res = None
-
-        response.content = json.dumps(res)
+        if 'data' in content and request.method == 'GET':
+            flattened_content = {}
```

Comment:
I could have declared it outside of the if/elif case once, but it's waste of memory for cases where the flow doesn't pass through either at runtime.

Also, I could have nested the if/else case like so:

```if 'data' in content:
          flattened_content = {}
          if request.method == 'GET':
              pass
          elif request.method == 'POST':
              pass```

but flat is better than nested.